### PR TITLE
Put Elab in Prelude and deprecate old-style tactics

### PR DESCRIPTION
**Pull request by david-christiansen**
This PR deprecates the old style of tactic proofs with a warning (but code still compiles, and they still work fine). It also brings elaborator reflection into the Prelude, so it doesn't need importing.

### CODE REVIEW

File: test/effects003/hangman.idr [hunk 1]
```diff
@@ -223,3 +223,7 @@ were caught by the type checker:
 
 main : IO ()
 main = run runGame
+
+-- Local Variables:
+-- idris-load-packages: ("effects")
```
Comment (Melvar) [inline test/effects003/hangman.idr]: Did you add this on purpose?

Comment (david-christiansen) [inline test/effects003/hangman.idr]: I did. It lets the code work interactively in idris-mode when debugging test failures.

**Comment by david-christiansen** [comment]
Hmm, I think I'd like to make it easier to remove tactic script deprecation warnings before I merge this. I'll add another command line argument.

**Comment by david-christiansen** [comment]
OK, once Travis is done, I think this is good enough for a merge.

**Comment by shlevy** [comment]
Is there a replacement for default implicits using elab scripts?

**Comment by Melvar** [comment]
`default` takes an expression, it doesn’t need to be a tactic script. A `tactics` block is an expression, and so is an invocation of `%runElab`, so you can use `{default (%runElab (do …)) prf : …} ->` and so on.

**Comment by shlevy** [comment]
Thanks!

**Pull request merged**